python/cudf/cudf/core/column/column.py

```diff
@@ -1343,7 +1343,16 @@ def _return_sentinel_column():
             )
 
         if dtype is None:
-            dtype = min_scalar_type(max(len(cats), na_sentinel), 8)
+            dtype = min_scalar_type(
+                max(
+                    len(cats),
+                    -1
+                    if isinstance(na_sentinel, cudf.Scalar)
+                    and na_sentinel.value is cudf.NA
+                    else na_sentinel,
+                ),
+                8,
+            )
```

Comment:
I wonder if it's cleaner for internal methods like this to always accept a `cudf.Scalar` when a scalar is expected.

(in the same way that we moved towards internal methods always accepting a `ColumnBase`, rather than any column-like object).

@vyasr do you have any thoughts?

Reply:
I think that's a better idea too, switched to accept only `cudf.Scalar`'s here.

Reply:
Yes, I would favor that approach.